### DISCUSSION TITLE
Issues with Scenes

### DISCUSSION THREAD

**Discussion by ancelin-charpentier** [cwapi3d/cwapi3dpython#228] [Q&A] [answered]
Hey there,

I got some issues with the scenes when they get created by an API.

My goal is to create a scene and assign some beams to it if the list I create for them is not empty.

- If the list is not empty = create a scene and assign the content of the list to it
- If the list is empty and the scene exist = delete the scene (Because it was created by a previous use of the API)
- If the list is empty and the scene does not exist = pass

My code work fine for the first try and then the scene never appears again even though the code get to the section that should create it.

See below a sample of the code: 

```
import cadwork
import scene_controller

not_nested_part_scene = "Not nested"

scene_list = scene_controller.get_scene_list()
print(scene_list)

if not_nested_part_scene in scene_list:
    scene_controller.delete_scene(not_nested_part_scene)
else:
    pass

not_nested = [123, 456, 789]

if not_nested:
    scene_controller.add_scene(not_nested_part_scene)
    scene_controller.add_elements_to_scene(not_nested_part_scene, not_nested)
```
 
Any of you have experienced that? I tried to refresh the display are other stuff but without any success.

Cheers

**Comment by NOPKGer** [comment, marked as answer]
hey, i don't know your problem like this but i think if you try it this way it should work.
You don't always have to recreate the scene, you can simply delete the existing elements and then insert the new ones.

````
import cadwork
import scene_controller

not_nested_part_scene = "Not nested"

scene_list = scene_controller.get_scene_list()
print(scene_list)

if not_nested_part_scene in scene_list:
    toRemove = scene_controller.get_elements_from_scene(not_nested_part_scene)
    scene_controller.remove_elements_from_scene(not_nested_part_scene,toRemove)
else:
    scene_controller.add_scene(not_nested_part_scene)

not_nested = [123, 456, 789]

if not_nested:
    scene_controller.add_elements_to_scene(not_nested_part_scene,not_nested)
````

**Reply by ancelin-charpentier** [reply]
Thanks @NOPKGer , it works fine as you described so I will stick to that. Feels like it is lagging when you try to create or delete the scene instead of using an existing one
